### PR TITLE
Add coverage testing

### PR DESCRIPTION
**Pull request by soltanmm**
Requires #2698.

Only enables coverage testing and JUnit testing results from `run_tests.py`. Does not actually make Travis CI aware of coverage testing. Does not yet enable building the Python extension with `gcov` (or similar measures to enable coverage testing of the C extension).

cc #943
(^ does *not* yet fix)

### CODE REVIEW

**Comment by nathanielmanistaatgoogle** [comment]
Looks mostly good although a proper review will have to wait for #2697 and #2698 to go in, right?

Is it worth drafting the "aggregate all Python tests together" change in a separate commit from the "add coverage" change?

**Comment by soltanmm** [comment]
I don't understand. The 'aggregate all Python tests together' bit and the 'add coverage' bit are already in two separate commits (745d848b6d12d8c5529c47ef0d2031012c641cdb and a0fb4a8e23985766c5bbb9d1ecf25c3747149fb6 respectively).

**Comment by nathanielmanistaatgoogle** [comment]
So they are; thanks for pointing that out.

**Comment by nathanielmanistaatgoogle** [comment]
This needs a rebase, yes?

File: src/python/grpcio_test/setup.py [hunk 1]
```diff
@@ -43,13 +52,25 @@
...
-    version='0.0.1',
+    version='0.10.0a0',
     packages=_PACKAGES,
     package_dir=_PACKAGE_DIRECTORIES,
     package_data=_PACKAGE_DATA,
-    install_requires=_INSTALL_REQUIRES
+    install_requires=_INSTALL_REQUIRES + _SETUP_REQUIRES,
+    setup_requires=_SETUP_REQUIRES,
+    cmdclass={
+        'test': commands.RunTests
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio_test/setup.py]: Comma at end of line? Perhaps a module-scope constant?

(Yes, this is me being fussy about little things.)

File: src/python/grpcio_test/setup.py [hunk 2]
```diff
@@ -43,13 +52,25 @@
...
+)
+
+_INSTALL_REQUIRES = (
+    'oauth2client>=1.4.7', 'grpcio>=0.10.0a0'
+)
 
 setuptools.setup(
     name='grpcio_test',
-    version='0.0.1',
+    version='0.10.0a0',
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio_test/setup.py]: Does this matter at all or are we keeping it in sync with grpcio's version number purely out of an aesthetic sense of symmetry?

Comment (soltanmm) [inline src/python/grpcio_test/setup.py]: Now that the tests and the main package are in separate projects I think they ought to be kept in sync. I probably should've made this change one PR earlier, actually.

**Comment by nathanielmanistaatgoogle** [comment]
LGTM and green enough on Travis - anything else to be done here before merge?

**Pull request merged**